src/main/java/org/example/c17inputoutput/ReadingAFileWithNIO2.java

```diff
@@ -0,0 +1,64 @@
+package org.example.c17inputoutput;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ReadingAFileWithNIO2 {
+    public static void main(String[] args) {
+
+        System.out.println(System.getProperty("file.encoding"));
+        System.out.println(Charset.defaultCharset());
+
+        Path path = Path.of("pathfile.txt");
+        try {
+            System.out.println(new String(Files.readAllBytes(path)));
+            System.out.println("----------------");
+            System.out.println(Files.readString(path));
+
+            Pattern p = Pattern.compile("(.{15})(.{3})(.{12})(.{8})(.{2}).*");
```

Comment:
Give this Pattern object a suggestive name.

Reply:
Done. New name is giveDimensionForColumn

Reply:
Same comment as above.

Reply:
groupsChar